cookbook/ui_customization/add_custom_information_to_a_field.rst

```diff
@@ -28,34 +28,59 @@ Let's say that we want to display the minimum and maximum values allowed for a n
-                                    '<span>Max value: ' + field.attribute.number_max + '</span>'
-                                );
-                            }
-
-                            //You can also disable the field with the setEditable method of the field
-                            //field.setEditable(false);
+                    var deferred = $.Deferred();
+                    event.promises.push(deferred);
+
+                    var field = event.field;
+                    if ('pim_catalog_number' === field.attribute.type) {
+                        if (!_.isNull(field.attribute.number_min) && !_.isEmpty(field.attribute.number_min)) {
+                            //To add html or a DOM element to a field, you can add the addElement method:
+                            // addElement: function (position, code, element)
+                            field.addElement(
+                                'footer',
+                                'number_min',
+                                '<span>Min value: ' + field.attribute.number_min + '</span>'
+                            );
                         }
-                    }.bind(this)).promise());
+
+                        if (!_.isNull(field.attribute.number_max) && !_.isEmpty(field.attribute.number_max)) {
+                            //To add html or a DOM element to a field, you can add the addElement method:
+                            // addElement: function (position, code, element)
+                            field.addElement(
+                                'footer',
+                                'number_max',
+                                '<span>Max value: ' + field.attribute.number_max + '</span>'
+                            );
+                        }
+
+                        //You can also disable the field with the setEditable method of the field
+                        //field.setEditable(false);
+                    }
+
+                    // After you successfully finished your computation, you have to resolve your
+                    // deferred Promise. Otherwise the field won't get rendered.
+                    // The resolve() can happen everywhere, even in a callback of another Promise.
+                    // 
+                    // Other Examples:
+                    //
+                    //     /**
+                    //      * Example waiting for another Promise to be resolved
+                    //      */
+                    //     FetcherRegistry
+                    //       .getFetcher('product')
+                    //       .fetch(this.parentProductId)
+                    //       .then(function (parentProduct) {
+                    //         deferred.resolve();
+                    //       });
+                    //
+                    //     /**
+                    //      * Example using a simple setTimeout
+                    //      */
+                    //     setTimeout(function () {
```

Comment:
Using timeouts is considered as a bad practice, so even if it's for the example maybe it's not a good idea to put it in our docs.